doc/api.rst

```diff
@@ -0,0 +1,251 @@
+=============================
+API: Interacting with backend
+=============================
+
+Available APIs
+--------------
+
+SQUAD has a set of APIs that allow to interact with it's backend. There
+are two main parts of the API
+
+- Native API
+  This is meant to provide main features of SQUAD (submitting results,
+  submitting CI test jobs)
+- REST API
+  Provides access to almost all poperties of core data model objects. Also
+  provides additional features that can be used to build alternative
+  frontends or automated tools.
+
+Native APIs
+-----------
+
+createbuild
+~~~~~~~~~~~
+
+**POST** /api/createbuild/<group_slug>/<project_slug>/<version_string>
+
+Creates Build object. Following parameters are accepted:
+
+- patch_source - string matching PatchSource.name
+- patch_baseline - version string matching Build.version
+- patch_id - string identifying the patched version (for example git commit ID)
+
+submit
+~~~~~~
+
+:ref:`result_submit_ref_label`.
+
+submitjob
+~~~~~~~~~
+
+:ref:`ci_job_ref_label`.
+
+watchjob
+~~~~~~~~
+
+:ref:`ci_watch_ref_label`.
+
+resubmit
+~~~~~~~~
+
+**GET** /api/resubmit/<job_id>
```

Comment:
I'm surprised this and forceresubmit are GET and not PUT

Reply:
I think it would work in both cases.

Reply:
It's out of scope for this PR, but I wouldn't expect a GET to make any changes.